atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionLockWatchingCacheView.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+// todo (gmaretic): decide if the logic for what can be cached should live here or in the cache itself
```

Comment:
I guess you mean the implementations of `tryCache(...)`? To me it feels like the cache should arbitrate such things - I'd perceive the role of a view as just letting you do `readCached` correctly - though there's room for discussion here.

Reply:
Yeah, makes sense -- made the change